components/script/dom/htmldetailselement.rs

```diff
@@ -116,30 +118,70 @@ impl HTMLDetailsElement {
             )
             .expect("Attaching UA shadow root failed");
 
-        let summary = HTMLSlotElement::new(local_name!("slot"), None, &document, None, can_gc);
+        let link_element = HTMLLinkElement::new(
+            local_name!("link"),
+            None,
+            &document,
+            None,
+            ElementCreator::ScriptCreated,
+            can_gc,
+        );
+        link_element.upcast::<Element>().set_attribute(
+            &local_name!("rel"),
+            AttrValue::String("stylesheet".to_owned()),
+            can_gc,
+        );
+        root.upcast::<Node>()
+            .AppendChild(link_element.upcast::<Node>(), can_gc)
+            .unwrap();
+
+        let details_stylesheet = parse_details_stylesheet(
```

Comment:
I think we need a mechanism to avoid reparsing the same stylesheet for every element.

But that can be left as TODO for now.

Reply:
Thanks, I will leave it as a todo for now.